pkg/models/apiv2.go

```diff
@@ -63,6 +63,24 @@ type CloudProviderSettings struct {
 	AzureSettings []*AzureSetting `json:"azureSettings,omitempty"`
 }
 
+// HasAWSCloudProviderSettings indicate if the AWSSetting object is not zero.
+// We need this because Instaclustr API returns an object of AWSSetting which is not empty,
+// but the fields are filled with null values.
+func (dc *CloudProviderSettings) HasAWSCloudProviderSettings() bool {
+	if dc == nil {
+		return false
+	}
+
+	zero := AWSSetting{}
+	for _, s := range dc.AWSSettings {
+		if *s == zero {
+			return false
+		}
+	}
+
+	return true
+}
```

Comment:
```
func (dc *CloudProviderSettings) HasAWSCloudProviderSettings() bool {
    if dc == nil || len(dc.AWSSettings) == 0 {
        return false
    }

    zero := AWSSetting{}
    for _, s := range dc.AWSSettings {
        if s != nil && *s != zero {
            return true
        }
    }

    return false
}
```

wdyt?

Reply:
Make sense use it to prevent potential panics if the current behaviour will be fixed. Thanks a lot.